src/main/java/cubicchunks/server/PlayerCubeMap.java

```diff
@@ -177,30 +178,15 @@ public void onConfigUpdate(CubicChunks.Config config) {
 	 * This method exists only because vanilla needs it. It shouldn't be used anywhere else.
 	 */
 	@Override
-	@Deprecated // BUG: no vertical filtering! TODO: do filtering and return a sort of 'ColumnView'
+	@Deprecated // Warning: Hacks! For vanilla use only! (WorldServer.updateBlocks())
 	public Iterator<Chunk> getChunkIterator() {
-		final Iterator<PlayerCubeMapColumnEntry> iterator = this.columnWatchers.valueCollection().iterator();
+		// GIVE TICKET SYSTEM FULL CONTROL
+		Iterator<Chunk> chunkIt = this.cubeCache.getLoadedChunks().iterator();
 		return new AbstractIterator<Chunk>() {
-			protected Chunk computeNext() {
-				while (iterator.hasNext()) {
-					PlayerCubeMapColumnEntry watcher = iterator.next();
-					Chunk column = watcher.getChunk();
-
-					// TODO: test Cubes, not Column
-
-					if (column == null) {
-						continue;
-					}
-					//columns that don't have light calculated or haven't been ticked
-					//have higher priority
-					if (!column.isLightPopulated() && column.isTerrainPopulated()) { 
-						return column;
-					}
-					if (!column.isChunkTicked()) {
-						return column;
-					}
-					//is there any non-spectator player within 128 blocks distance?
-					if (watcher.hasPlayerMatchingInRange(128.0D, NOT_SPECTATOR)) {
+			@Override protected Chunk computeNext() {
+				while (chunkIt.hasNext()){
+					Column column = (Column)chunkIt.next();
+					if(column.shouldTick()){ // shouldTick is true when there Cubes with tickets the request to be ticked
```

Comment:
PlayerCubeMap.getChunkIterator() is just being used like an event hook now...
Tickets now control when Cubes should be ticked, (PlayerCubeMap uses tickets now so its ok)